### PR TITLE
Adding in ability to update Hyperdisk IOPS without recreation

### PR DESCRIPTION
**Pull request by NA2047**
Adding in Support for in-place update of Provisioned IOPS  for Hyperdisk SKUs on google_compute_disk resources https://cloud.google.com/compute/docs/disks/hyperdisks
Currently the only PD SKU that is supports in-place updates to Provisioned IOPS 

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
Adding in ability to update Hyperdisk PD IOPS without recreation
```

### CODE REVIEW

File: mmv1/products/compute/Disk.yaml [hunk 1]
```diff
@@ -431,6 +432,10 @@ properties:
       Indicates how many IOPS must be provisioned for the disk.
     required: false
     default_from_api: true
+    description: |
```
Comment (slevenick) [inline mmv1/products/compute/Disk.yaml:435]: This duplicates the above description, and will likely cause a compilation error

File: mmv1/products/compute/Disk.yaml [hunk 2]
```diff
@@ -431,6 +432,10 @@ properties:
...
+     Update currently only supported by hyperdisk skus, allowing for an update of IOPS every 4 hours
+    update_verb: :PATCH
```
Comment (slevenick) [inline mmv1/products/compute/Disk.yaml:437]: Does adding an update_verb on a resource marked immutable work? Were you able to successfully update in-place a disk with a provider built off of this?

**Review by slevenick — CHANGES_REQUESTED** [review]
Can you add a test that updates this field in-place? It should be possible with three separate configs

1. Set up the disk with IOPS set and lifecycle {prevent_destroy = true} set
2. Modify the IOPS with the lifecycle block still in place
3. Same as 2 but remove the lifecycle block to allow cleanup

File: mmv1/products/compute/Disk.yaml (hunk 2 above)
Comment (NA2047) [reply mmv1/products/compute/Disk.yaml:436]: It works I have tested it successfully I have built the provider successfully and it updates inplace no issue, while maintaining the ForceNew on all other PDs except Hyper Disk

**Comment by slevenick** [comment]
> > Can you add a test that updates this field in-place? It should be possible with three separate configs
> > 
> > 1. Set up the disk with IOPS set and lifecycle {prevent_destroy = true} set
> > 2. Modify the IOPS with the lifecycle block still in place
> > 3. Same as 2 but remove the lifecycle block to allow cleanup
> 
> Added the Tests @slevenick I dont know why the test is saying undefined: ResourceComputeDisk it builds and works locally for me

It's failing to build a secondary component of the provider. Can you add the following tags to the files in the constants folder: `<% unless compiler == "terraformvalidator-codegen" -%>` There should be a few examples of that in there.

**Review by slevenick — CHANGES_REQUESTED** [review]
I believe you need <% unless compiler == "terraformvalidator-codegen" -%> tags in constants  to make the downstreams compile correctly. Some external tools try to compile this too

**Comment by NA2047** [comment]
> I believe you need <% unless compiler == "terraformvalidator-codegen" -%> tags in constants to make the downstreams compile correctly. Some external tools try to compile this too

Hey @slevenick  Thanks for the help on this I'm getting some issues with my tests, which have indicated to be an issue with the `prevent_destroy` and not being able to clean up the resources

**Review by slevenick — CHANGES_REQUESTED** [review]
Looks like it's failing with:
    vcr_utils.go:147: Step 3/6 error: Error running pre-apply refresh: exit status 1
        
        Error: Instance cannot be destroyed
        
          on terraform_plugin_test.tf line 2:
           2: 	resource "google_compute_disk" "foobar" {


So the resource is trying to recreate during step3 which should only change the IOPS. Looks like you're recalculating the random suffix independently for each context so the name of the resource is changing. Try having one random suffix that you share across the contexts

**Comment by NA2047** [comment]
> Looks like it's failing with: vcr_utils.go:147: Step 3/6 error: Error running pre-apply refresh: exit status 1
> 
> ```
>     Error: Instance cannot be destroyed
>     
>       on terraform_plugin_test.tf line 2:
>        2: 	resource "google_compute_disk" "foobar" {
> ```
> 
> So the resource is trying to recreate during step3 which should only change the IOPS. Looks like you're recalculating the random suffix independently for each context so the name of the resource is changing. Try having one random suffix that you share across the contexts

Looks like its passing now! @slevenick

**Review by slevenick — APPROVED** [review]

**Review by slevenick — CHANGES_REQUESTED** [review]
Looks good to me, but is a conflict in one of the files now. Can you fix that and I'll merge after the tests run again?

**Comment by slevenick** [comment]
/gcbrun

I'm a little confused as to what's going on here. It seems you've caught some errors from an unrelated change that was merged in here and it's causing problems in a downstream repository. Maybe building again will work 🤞

File: mmv1/templates/terraform/resource_definition/disk.erb [hunk 3]
```diff
@@ -1,2 +1,3 @@
 CustomizeDiff: customdiff.All(
-  customdiff.ForceNewIfChange("size", IsDiskShrinkage)),
+  customdiff.ForceNewIfChange("size", IsDiskShrinkage)
```
Comment (slevenick) [inline mmv1/templates/terraform/resource_definition/disk.erb:2]: Looks like you're missing a comma here to separate the arguments

**Review by slevenick — CHANGES_REQUESTED** [review]
Small typo

**Comment by NA2047** [comment]
> Small typo

Fixed and looks like its passing @slevenick

**Review by slevenick — APPROVED** [review]

**Pull request merged**